### ISSUE TITLE
Maximum iterations

### ISSUE THREAD

**Issue by minxueric** [Dongwon-Lee/lsgkm#2]
Is there any way to set the maximum iterations of the training procedure? 
Sometimes, the training loss is decreasing incredibly slowly. It seems it is unnecessary to keep optimizing the loss function which doesn't bring us much improvement.
